File: FinalProject_OOP_C#/Door.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FinalProject_OOP
{
    public abstract class Door:ListDoor_Interface
    {
        protected string sDoor_ID;
        protected string sDoor_IDStall;
        protected string sDoor_Status;
        protected List<Device> m_ElectricGateList;
        protected List<Device> m_ElevatorList;
        public string Door_ID
        {
            get { return this.sDoor_ID; }
            set { this.sDoor_ID = value; }
        }
         public string Door_IDStall
        {
            get { return this.sDoor_IDStall; }
            set { this.sDoor_IDStall=value; }
        }
        public string Door_Status
        {
            get { return this.sDoor_Status; }
            set
            {
                this.sDoor_Status = value;
            }
        }
        
        protected Door()
        {
            this.Door_ID = "";
            this.Door_IDStall = "";
            this.Door_Status = "work";
        }
        ~Door() { }

        protected Door( string door_ID, string door_IDStall, string door_Status)
        {
            this.Door_ID = door_ID;
            this.Door_IDStall = door_IDStall;
            this.Door_Status = door_Status;
        }
        public void Input(string door_ID, string door_IDStall, string door_Status)
        {
            this.Door_ID = door_ID;
            this.Door_IDStall=door_IDStall;
            this.Door_Status=door_Status;
        }
        public virtual void Input()
        {
            Console.WriteLine("ID door:"); 
            this.Door_ID = Console.ReadLine();
            Console.Write("ID stall of door:");
            this.Door_IDStall = Console.ReadLine();
            Console.WriteLine("Situation:");
            this.Door_Status = Console.ReadLine();
        }
        public virtual void Ouput()
        {
            Console.WriteLine(this.Door_ID + "   |  " + this.Door_IDStall + "  |        " + this.Door_Status+"       |");
        }
        public abstract void Docfile(string path1,string path2);
        public abstract void Ouput1();
        public string DoorList_FindingDoor_address(string name)
        {
            name = name.ToLower();
            if (name == "door 1" || name == "door1") return Door_List.Door_1.Door_ID;
            else if (name == "door 2" || name == "door2") return Door_List.Door_2.Door_ID;
            else if (name == "door 3" || name == "door3") return Door_List.Door_3.Door_ID;
            else if (name == "door 4" || name == "door4") return Door_List.Door_4.Door_ID;
            else if (name == "door 5" || name == "door5") return Door_List.Door_5.Door_ID;
            else return "ERROR";
        }
        public void OuputListDoor(string ID)
        {
            foreach (Door door in SystemPark.List_Doors)
            {
                if (door.Door_ID == ID)
                {
                    door.Ouput1();
                    return;
                }
            }
            Console.WriteLine("No information!!!");
        }
        
        /*
        string DoorList_FindingDoor_address(string name)
        {
            name = name.ToLower();
            if (name == "door 1" || name == "door1") return Door_List.Door_1.Door_ID;
            else if (name == "door 2" || name == "door2") return Door_List.Door_2.Door_ID;
            else if (name == "door 3" || name == "door3") return Door_List.Door_3.Door_ID;
            else if (name == "door 4" || name == "door4") return Door_List.Door_4.Door_ID;
            else if (name == "door 5" || name == "door5") return Door_List.Door_5.Door_ID;
            else return "ERROR";
        }
        void OuputListDoor(string ID)
        { 
            foreach(Door door in SystemPark.List_Doors) 
            {
                if (door.Door_ID == ID)
                {
                    door.Ouput();
                    return;
                }
            }
            Console.WriteLine("No information!!!");
        }
        */
    }
}

File: FinalProject_OOP_Java/src/Door/Door.java
package Door;

import java.io.FileNotFoundException;
import java.util.List;

import Device.Device;
import System.ListDoor_Interface;
import System.SystemPark;

public abstract class Door implements ListDoor_Interface
{
	protected String sDoor_ID;
    protected String sDoor_IDStall;
    protected String sDoor_Status;
    protected List<Device> m_ElectricGateList;
    protected List<Device> m_ElevatorList;
    
    public String getsDoor_ID() {
		return sDoor_ID;
	}
	public void setsDoor_ID(String sDoor_ID) {
		this.sDoor_ID = sDoor_ID;
	}
	public String getsDoor_IDStall() {
		return sDoor_IDStall;
	}
	public void setsDoor_IDStall(String sDoor_IDStall) {
		this.sDoor_IDStall = sDoor_IDStall;
	}
	public String getsDoor_Status() {
		return sDoor_Status;
	}
	public void setsDoor_Status(String sDoor_Status) {
		this.sDoor_Status = sDoor_Status;
	}
	public List<Device> getM_ElectricGateList() {
		return m_ElectricGateList;
	}
	public void setM_ElectricGateList(List<Device> m_ElectricGateList) {
		this.m_ElectricGateList = m_ElectricGateList;
	}
	public List<Device> getM_ElevatorList() {
		return m_ElevatorList;
	}
	public void setM_ElevatorList(List<Device> m_ElevatorList) {
		this.m_ElevatorList = m_ElevatorList;
	}
	protected Door()
    {
        this.setsDoor_ID("");
        this.setsDoor_IDStall("");
        this.setsDoor_Status("work");
    }
	protected Door( String door_ID, String door_IDStall, String door_Status)
    {
		this.setsDoor_ID(door_ID);
		this.setsDoor_IDStall(door_IDStall);
		this.setsDoor_Status(door_Status);
    }
	
	public void Input(String door_ID, String door_IDStall, String door_Status)
    {
		this.setsDoor_ID(door_ID);
        this.setsDoor_IDStall(door_IDStall);
        this.setsDoor_Status(door_Status);
    }
	
	public void Ouput()
    {
        System.out.println(this.sDoor_ID + "   |  " + this.sDoor_IDStall + "  |        " + this.getsDoor_Status()+"       |");
    }
	public abstract void Docfile(String path1,String path2) throws FileNotFoundException;
	public abstract void Ouput1();
	public String DoorList_FindingDoor_address(String name)
    {
        name=name.toLowerCase();
        if (name == "door 1" || name == "door1") return Door_List.Door_1.getsDoor_ID();
        else if (name.compareTo("door 2")==0  || name.compareTo("door2")==0) return Door_List.Door_2.getsDoor_ID();
        else if (name.compareTo("door 3")==0 || name.compareTo("door3")==0) return Door_List.Door_3.getsDoor_ID();
        else if (name.compareTo("door 4")==0 || name.compareTo("door4")==0) return Door_List.Door_4.getsDoor_ID();
		else if (name.compareTo("door 5")==0 || name.compareTo("door5")==0) return Door_List.Door_5.getsDoor_ID();
        else return "ERROR";
    }
	
	 public void OuputListDoor(String ID)
     {
		 List<Door> ls= SystemPark.List_Doors;
		 for (int i=0; i<ls.size(); i++)
         {
             if (ls.get(i).getsDoor_ID().compareTo(ID)==0)
             {
            	 ls.get(i).Ouput1();
            	 return;
             }
         }
         System.out.println("No information!!!");
     }
}

